file_transfer_agent_test.go

```diff
@@ -532,3 +532,69 @@ func TestUpdateMetadataWithPresignedUrlError(t *testing.T) {
 		t.Fatalf("unexpected error code. expected: %v, got: %v", ErrCommandNotRecognized, driverErr.Number)
 	}
 }
+
+func TestUploadWhenFilesystemReadOnlyError(t *testing.T) {
+	var err error
+	roPath := t.TempDir()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Set the temp directory to read only
+	err = os.Chmod(roPath, 0444)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	info := execResponseStageInfo{
+		Location:     "gcs-blob/storage/users/456/",
+		LocationType: "GCS",
+	}
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Make sure that the test uses read only directory
+	originalTmpDir, envPresent := os.LookupEnv("TMPDIR")
+	os.Setenv("TMPDIR", roPath)
+	if envPresent {
+		defer os.Setenv("TMPDIR", originalTmpDir)
+	} else {
+		defer os.Unsetenv("TMPDIR")
+	}
+
+	uploadMeta := fileMetadata{
+		name:              "data1.txt.gz",
+		stageLocationType: "GCS",
+		noSleepingTime:    true,
+		client:            gcsClient,
+		sha256Digest:      "123456789abcdef",
+		stageInfo:         &info,
+		dstFileName:       "data1.txt.gz",
+		srcFileName:       path.Join(dir, "/test_data/data1.txt"),
+		overwrite:         true,
+		options: &SnowflakeFileTransferOptions{
+			MultiPartThreshold: dataSizeThreshold,
+		},
+	}
+
+	sfa := &snowflakeFileTransferAgent{
+		sc:                nil,
+		commandType:       uploadCommand,
+		command:           "put file:///tmp/test_data/data1.txt @~",
+		stageLocationType: gcsClient,
+		fileMetadata:      []*fileMetadata{&uploadMeta},
+	}
+
+	// Enable parallel transfer
```

Comment:
Does it really? I'd expect that setting this value sets maximum number of parallel uploads :)

Reply:
Right, updated the comment.